texar/modules/connectors/connectors.py

```diff
@@ -0,0 +1,866 @@
+                self.register_parameter(dstr_attr, dstr_param)
+
+        input_feature = np.prod(mlp_input_size)
+        self._linear_layer = nn.Linear(
+            input_feature, _sum_output_size(output_size))
+
+    @staticmethod
+    def default_hparams() -> dict:
+        r"""Returns a dictionary of hyperparameters with default values.
+
+        .. code-block:: python
+
+            {
+                "activation_fn": "texar.core.layers.identity",
+                "name": "reparameterized_stochastic_connector"
+            }
+
+        Here:
+
+        "activation_fn" : str
+            The activation function applied to the outputs of the MLP
+            transformation layer. Can be a function,
+            or its name or module path.
+        "name" : str
+            Name of the connector.
+        """
+        return {
+            "activation_fn": "texar.core.layers.identity",
+            "name": "reparameterized_stochastic_connector"
+        }
+
+    def forward(self,    # type: ignore
+                num_samples: Optional[Union[int, torch.Tensor]] = None,
+                transform: bool = True) -> Tuple[Any, Any]:
+        r"""Samples from a distribution and optionally performs transformation
+        with an MLP layer.
+        The distribution must be reparameterizable, i.e.,
+        :python:`Distribution.has_rsample == True`.
+
+        Args:
+            num_samples (optional): An ``int`` or ``int`` ``Tensor``.
+                Number of samples to generate. If not given,
+                generate a single sample. Note that if batch size has
+                already been included in :attr:`distribution`'s dimensionality,
+                :attr:`num_samples` should be left as ``None``.
+            transform (bool): Whether to perform MLP transformation of the
+                distribution samples. If ``False``, the structure/shape of a
+                sample must match :attr:`output_size`.
+
+
+        :returns:
+            If attr:`transform` is ``True``, returns a ``tuple``
+            (attr:`output`, attr:`sample`), where
+            - output: A ``Tensor`` or a (nested) ``tuple`` of Tensors with
+              the same structure and size of :attr:`output_size`.
+              The batch dimension equals :attr:`num_samples` if specified,
+              or is determined by the distribution dimensionality.
+            - sample: The sample from the distribution, prior to transformation.
+
+            Otherwise, returns a ``Tensor`` attr:`sample`, where
+            - sample: The sample from the distribution, prior to transformation.
+
+        Raises:
+            ValueError: If distribution cannot be reparametrized.
+            ValueError: The output does not match :attr:`output_size`.
+        """
+
+        distribution = utils.check_or_get_instance(
+            self._dstr_type, self._dstr_kwargs,
+            ["torch.distributions", "torch.distributions.multivariate_normal",
```

Comment:
Do we need to include `torch.distributions.multivariate_normal`? Isn't `MultivariateNormal` included in `torch.distributions`?